pkg/api/handlers/compat/networks.go

```diff
@@ -56,45 +86,46 @@ func InspectNetwork(w http.ResponseWriter, r *http.Request) {
 		utils.NetworkNotFound(w, name, err)
 		return
 	}
-	report, err := convertLibpodNetworktoDockerNetwork(runtime, &net, changed)
+	statuses, err := getContainerNetStatuses(runtime)
+	if err != nil {
+		utils.InternalServerError(w, err)
+		return
+	}
+	report, err := convertLibpodNetworktoDockerNetwork(runtime, statuses, &net, changed)
 	if err != nil {
 		utils.InternalServerError(w, err)
 		return
 	}
 	utils.WriteResponse(w, http.StatusOK, report)
 }
 
-func convertLibpodNetworktoDockerNetwork(runtime *libpod.Runtime, network *nettypes.Network, changeDefaultName bool) (*types.NetworkResource, error) {
-	cons, err := runtime.GetAllContainers()
-	if err != nil {
-		return nil, err
-	}
-	containerEndpoints := make(map[string]types.EndpointResource, len(cons))
-	for _, con := range cons {
-		data, err := con.Inspect(false)
-		if err != nil {
-			if errors.Is(err, define.ErrNoSuchCtr) || errors.Is(err, define.ErrCtrRemoved) {
-				continue
-			}
-			return nil, err
-		}
-		if netData, ok := data.NetworkSettings.Networks[network.Name]; ok {
+func convertLibpodNetworktoDockerNetwork(runtime *libpod.Runtime, statuses []containerNetStatus, network *nettypes.Network, changeDefaultName bool) (*types.NetworkResource, error) {
```

Comment:
there's no case where we have an error returned

Reply:
fixed